File: source_funcs/go_transcoder/CHECK_TWO_GIVEN_CIRCLES_TOUCH_INTERSECT/CHECK_TWO_GIVEN_CIRCLES_TOUCH_INTERSECT_emit.go
package main

//go:export f_gold
func f_gold(x1 int, y1 int, x2 int, y2 int, r1 int, r2 int) int {
	var (
		distSq   int = (x1-x2)*(x1-x2) + (y1-y2)*(y1-y2)
		radSumSq int = (r1 + r2) * (r1 + r2)
	)
	if distSq == radSumSq {
		return 1
	} else if distSq > radSumSq {
		return -1
	} else {
		return 0
	}
}

File: rust_funcs/go_transcoder/CHECK_TWO_GIVEN_CIRCLES_TOUCH_INTERSECT/CHECK_TWO_GIVEN_CIRCLES_TOUCH_INTERSECT_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


#[no_mangle]
fn f_gold(x1: i32, y1: i32, x2: i32, y2: i32, r1: i32, r2: i32) -> i32 {
    let x1 = x1 as f32;
    let y1 = y1 as f32;
    let x2 = x2 as f32;
    let y2 = y2 as f32;
    let r1 = r1 as f32;
    let r2 = r2 as f32;
    let dist_sq = (x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2);
    let rad_sum_sq = (r1 + r2) * (r1 + r2);
    if dist_sq == rad_sum_sq {1} 
    else if dist_sq > rad_sum_sq {-1}
    else {0}
}
